lib/ui/viewer/file/file_details_widget.dart

```diff
@@ -231,36 +230,32 @@ class _FileDetailsWidgetState extends State<FileDetailsWidget> {
     );
   }
 
-  bool _hasLocationData() {
+  Future<bool> _hasLocationData() async {
     final fileLocation = widget.file.location;
-    final hasLocation = (fileLocation != null &&
+    bool hasLocation = (fileLocation != null &&
             fileLocation.latitude != null &&
             fileLocation.longitude != null) &&
         (fileLocation.latitude != 0 || fileLocation.longitude != 0);
+
+    //This code is for updating the location of files in which location data is
+    //missing and the EXIF has location data. This is only happens for a
+    //certain specific minority of devices.
+    if (!hasLocation) {
+      hasLocation = await _setLocationDataFromExif();
+    }
     return hasLocation;
   }
 
-  void _generateExifForLocation(Map<String, IfdTag> exif) {
-    if (exif["GPS GPSLatitude"] != null) {
-      _exifData["lat"] = exif["GPS GPSLatitude"]!
-          .values
-          .toList()
-          .map((e) => ((e as Ratio).numerator / e.denominator))
-          .toList();
-    }
-    if (exif["GPS GPSLongitude"] != null) {
-      _exifData["long"] = exif["GPS GPSLongitude"]!
-          .values
-          .toList()
-          .map((e) => ((e as Ratio).numerator / e.denominator))
-          .toList();
-    }
-    if (exif["GPS GPSLatitudeRef"] != null) {
-      _exifData["latRef"] = exif["GPS GPSLatitudeRef"].toString();
-    }
-    if (exif["GPS GPSLongitudeRef"] != null) {
-      _exifData["longRef"] = exif["GPS GPSLongitudeRef"].toString();
+  Future<bool> _setLocationDataFromExif() async {
```

Comment:
Instead of reading the exif again here, can we add logic at the place where we are reading exif for showing fileInfo?
In that, if file.location is missing but exif have location, we can attach the location.
Also, if the file is already uploaded, we can update the pubMetadata?